components/script/dom/htmliframeelement.rs

```diff
@@ -162,11 +162,14 @@ impl HTMLIFrameElement {
         if load_data.url.scheme() == "javascript" {
             let window_proxy = self.GetContentWindow();
             if let Some(window_proxy) = window_proxy {
+                let global = window_proxy.global();
+                if global.should_navigation_request_be_blocked(&load_data.url) {
```

Comment:
I think we actually want to use `self.owner_global()` for the CSP check here, which corresponds to the window that contains the iframe. This is where using the policy container stored in the LoadData would also be possible, if we did that.

Reply:
In fact, it seems like tests want this to be the document instead: https://github.com/web-platform-tests/wpt/blob/5726251d00e45c1ef06b0894dd843a93359c8f13/content-security-policy/navigation/to-javascript-url-script-src.html#L12 This is strange, given all tests listen on the `window`, but for now I want to follow the test.

Turns out that the resource should also be `inline` instead (filed a spec issue https://github.com/w3c/webappsec-csp/issues/719).

Lastly, I tried to implement the `policy_container` logic, but that doesn't seem to work yet unfortunately. I pushed my changes to the other PR and updated this one with the document.